WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionDeeplinkActivity.java

```diff
@@ -108,34 +123,36 @@ protected void onStart() {
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
-        if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK && source == Source.STATS) {
-            showStats();
+        if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK) {
+            trackResultAndFinish();
         } else {
-            finish();
+            finishAndGoBackToSource();
         }
     }
 
-    private void showStats() {
-        if (!TextUtils.isEmpty(reason)) {
-            ToastUtils.showToast(this, reason);
-            finish();
-        } else {
-            SiteModel site = (SiteModel) getIntent().getSerializableExtra(SITE);
-            mDispatcher.dispatch(SiteActionBuilder.newFetchSiteAction(site));
-        }
+    private void reloadSite() {
+        //Reloads the site
+        SiteModel site = (SiteModel) getIntent().getSerializableExtra(SITE);
+        mDispatcher.dispatch(SiteActionBuilder.newFetchSiteAction(site));
     }
 
     @Override
     public void onBackPressed() {
         super.onBackPressed();
-        finish();
+        finishAndGoBackToSource();
     }
 
     @SuppressWarnings("unused")
     @Subscribe(threadMode = ThreadMode.MAIN)
     public void onSiteChanged(SiteStore.OnSiteChanged event) {
-        SiteModel site = (SiteModel) getIntent().getSerializableExtra(SITE);
-        ActivityLauncher.viewBlogStats(this, site);
+        finishAndGoBackToSource();
+    }
+
+    private void finishAndGoBackToSource() {
+        if (source != null && source == Source.STATS) {
```

Comment:
I don't think we need to do a null check here, since if it's null the other equality will fail anyway.